cargo-dist/src/lib.rs

```diff
@@ -395,13 +396,133 @@ fn write_checksum(checksum: &str, src_path: &Utf8Path, dest_path: &Utf8Path) ->
+        desired_env.push(("CMAKE_LIBRARY_PATH".to_owned(), value.to_string()))
+    }
+    let mut paths = vec![];
+    // For each listed dependency, add it to the PATH
+    if let Some(optprefix) = environment.get("HOMEBREW_OPT") {
+        // Comma-separated list of all dependencies in the recursive tree
+        if let Some(dependencies) = environment.get("HOMEBREW_DEPENDENCIES") {
+            for dep in dependencies.split(',') {
+                let short_name = dep.split('/').next().unwrap();
+                let pkg_opt = format!("{optprefix}/{short_name}");
+
+                // Not every package will have a /bin or /sbin directory,
+                // but it's safe to add both to the PATH just in case.
+                paths.push(format!("{pkg_opt}/bin"));
+                paths.push(format!("{pkg_opt}/sbin"));
+            }
+        }
+    }
+    if !paths.is_empty() {
+        let our_path = env!("PATH");
+        let desired_path = format!("{our_path}:{}", paths.join(":"));
+
+        desired_env.insert(0, ("PATH".to_owned(), desired_path));
+    }
+
+    desired_env
+}
+
+// Similar to the above, we read Homebrew's recursive dependency tree and
+// then append link flags to cargo-dist's rustflags.
+// These ensure that Rust can find C libraries that may exist within
+// each package's prefix.
+fn determine_brew_rustflags(base_rustflags: &str, environment: &BTreeMap<&str, &str>) -> String {
+    let mut rustflags = base_rustflags.to_owned();
+    // For each listed dependency, add it to CFLAGS/LDFLAGS
+    if let Some(optprefix) = environment.get("HOMEBREW_OPT") {
+        if let Some(dependencies) = environment.get("HOMEBREW_DEPENDENCIES") {
+            for dep in dependencies.split(',') {
+                let short_name = dep.split('/').next().unwrap();
+                let pkg_opt = format!("{optprefix}/{short_name}");
+
+                // Note that this path might not actually exist; not every
+                // package contains libraries. However, it's safe to
+                // append this flag anyway; Rust passes it on to the
+                // compiler/linker, which tolerate missing directories
+                // just fine.
+                rustflags = format!("{rustflags} -L{pkg_opt}/lib");
+            }
+        }
+    }
+
+    rustflags
+}
+
 /// Build a cargo target
 fn build_cargo_target(dist_graph: &DistGraph, target: &CargoBuildStep) -> Result<()> {
     eprint!(
         "building cargo target ({}/{}",
         target.target_triple, target.profile
     );
 
+    let mut rustflags = target.rustflags.clone();
+    let mut desired_extra_env = vec![];
+    let skip_brewfile = env::var("DO_NOT_USE_BREWFILE").is_ok();
+    if let Some(brew) = &dist_graph.tools.brew {
+        if Utf8Path::new("Brewfile").exists() && !skip_brewfile {
+            // Uses `brew bundle exec` to just print its own environment,
+            // allowing us to capture what it generated and decide what
+            // to do with it.
```

Comment:
Does this mean that the code to invoke `brew bundle exec` in `ci.yml.j2` can be removed? (If so, please do)

Reply:
Yep! Thanks for the reminder, I actually missed adding that when I wrote this as quickly as I did.